doc/getting_started/toolchain_custom_cmake.rst

```diff
@@ -3,25 +3,29 @@
 Custom CMake Toolchains
 #######################
 
-To use a custom toolchain defined in an external CMake file, export the
-following environment variables:
+To use a custom toolchain defined in an external CMake file, :ref:`set these
+environment variables <env_vars>`:
 
-.. code-block:: console
+- Set :envvar:`ZEPHYR_TOOLCHAIN_VARIANT` to your toolchain's name
+- Set :envvar:`TOOLCHAIN_ROOT` to the path to the directory containing your
+  toolchain's CMake configuration files.
+
+Zephyr will then include the toolchain cmake files located in the
+:file:`TOOLCHAIN_ROOT` directory:
 
-   # Linux and macOS
-   export ZEPHYR_TOOLCHAIN_VARIANT=<toolchain name>
-   export TOOLCHAIN_ROOT=<path to toolchain>
+- :file:`cmake/toolchain/generic.cmake`: configures the toolchain for "generic"
+  use (mostly to run the C preprocessor on the generated :ref:`device-tree`
+  file).
+- :file:`cmake/toolchain/target.cmake`: configures the toolchain for use
+  building Zephyr and your application's source code.
 
-   # Windows
-   set ZEPHYR_TOOLCHAIN_VARIANT=<toolchain name>
-   set TOOLCHAIN_ROOT=<path to toolchain>
+See the zephyr files :zephyr_file:`cmake/generic_toolchain.cmake` and
+:zephyr_file:`cmake/target_toolchain.cmake` for more details on what your
+:file:`generic.cmake` and :file:`target.cmake` files should contain.
 
-You can also set them as CMake variables when generating a build
-system for a Zephyr application, like so:
+You can also set ``ZEPHYR_TOOLCHAIN_VARIANT`` and ``TOOLCHAIN_ROOT`` as CMake
+variables when generating a build system for a Zephyr application, like so:
```

Comment:
A bit "late to the party"; just realized it would be great to add a small `cmake -C project1_params.cmake` example here. Because `-C` is terribly misnamed "preload cache" in cmake documentation I kept missing it. Eventually I found it's perfect for doing exactly this while scaling so much better: long lists, version-controlled configurations, etc.

Reply:
I've added a note in a follow-on commit touching just this file.